docs/advanced/csidriver.md

```diff
@@ -0,0 +1,63 @@
+
+## Create Harvester cluster
+
+Harvester's operating system follows an immutable design, meaning that most OS files revert to their pre-configured state after a reboot. Therefore, you might need to perform additional configurations before installing the Harvester cluster for third-party CSI drivers.
+
+Some CSI drivers require additional persistent paths on the host. You can add these paths to [`os.persistentStatePaths`](../install/harvester-configuration.md#ospersistent_state_paths).
+
+Some CSI drivers require additional software packages on the host. You can install these packages with [`os.after_install_chroot_commands`](../install/harvester-configuration.md#osafter_install_chroot_commands).
+
+:::note
+
+Upgrading Harvester causes the changes to the OS in the `after-install-chroot` stage to be lost. You must also configure the `after-upgrade-chroot` to make your changes persistent across an upgrade. Refer to [Runtime persistent changes](https://rancher.github.io/elemental-toolkit/docs/customizing/runtime_persistent_changes/) before upgrading Harvester.
+
+:::
+
+## Install the CSI driver
+
+After installing the Harvester cluster is complete, refer to [How can I access the kubeconfig file of the Harvester cluster?](../faq.md#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) to get the kubeconfig of the cluster.
+
+With the kubeconfig of the Harvester cluster, you can install the third-party CSI drivers into the cluster by following the installation instructions for each CSI driver. You must also refer to the CSI driver documentation to create the StorageClass and VolumeSnapshotClass in the Harvester cluster.
+
+## Configure Harvester Cluster
+
+Before you can make use of Harvester's **Backup & Snapshot** features, you need to set up some essential configurations through the Harvester [csi-driver-config](../advanced/settings.md#csi-driver-config) setting. Follow these steps to make these configurations: 
+
+1. Login to the Harvester UI, then navigate to **Advanced** > **Settings**.
+1. Find and select **csi-driver-config**, and then select **⋮** > **Edit Setting** to access the configuration options.
+1. Set the **Provisioner** to the third-party CSI driver in the settings.
+1. Next, Configure the **Volume Snapshot Class Name**. This setting points to the name of the `VolumeSnapshotClass` used for creating volume snapshots or VM snapshots.
+1. Similarly, Configure the **Backup Volume Snapshot Class Name**. This corresponds to the name of the `VolumeSnapshotClass` responsible for creating VM backups.
+
+## Use the CSI driver
+
+After successfully configuring these settings, you can utilize the third-party StorageClass. You can apply the third-party StorageClass when creating an empty volume or adding a new block volume to a VM, enhancing your Harvester cluster's storage capabilities.
+
+With these configurations in place, your Harvester cluster is ready to make the most of the third-party storage integration.
```

Comment:
Consider adding a new section of `## References` for KB articles.

Reply:
Added